vanetza/security/validity_restriction.hpp

```diff
@@ -35,6 +36,47 @@ class Duration
         return m_raw;
     }
 
+    /**
+     * Returns the duration unit.
+     */
+    Units unit() const
+    {
+        return static_cast<Units>(m_raw >> 13);
+    }
+
+    uint16_t value() const
+    {
+        return m_raw & 0xCFFFF; // mask upper 3 bit
+    }
+
+    /**
+     * Returns the duration in seconds.
+     */
+    Time64 to_seconds() const
+    {
+        // see section 4.2.17 of TS 103 097 v1.2.1
+        switch (unit()) {
+            case Units::Seconds:
+                return value();
+
+            case Units::Minutes:
+                return value() * 60;
+
+            case Units::Hours:
+                return value() * 3600;
+
+            case Units::Sixty_Hour_Blocks:
+                return value() * 216000;
+
+            case Units::Years:
+                return value() * 31556925;
+
+            default:
+                // undefined, let's interpret it as a duration of 0, which will fail during validation
+                return 0;
+        }
+    }
```

Comment:
Larger code segments should go to the source file, i.e. everything except simple getter/setter and template code.